experimental/dds/tree2/src/core/tree/anchorSet.ts

```diff
@@ -608,19 +651,28 @@ export class AnchorSet implements ISubscribable<AnchorSetRootEvents> {
 						pathVisitor.onInsert(upPath, content);
 					}
 				}
-				this.moveChildren(content.length, undefined, {
-					parent,
-					parentField,
-					parentIndex: start,
-				});
+
+				this.offsetChildren(
+					{
+						parent,
+						parentField,
+						parentIndex: start,
+					},
+					content.length,
+				);
 			},
 			onMoveOut: (start: number, count: number, id: Delta.MoveId): void => {
 				assert(parentField !== undefined, 0x3a9 /* Must be in a field to move out */);
 				maybeWithNode(
 					(p) => p.events.emit("childrenChanging", p),
 					() => this.events.emit("childrenChanging", this),
 				);
-				moveTable.set(id, { parent, parentField, parentIndex: start });
+
+				const fieldKey = this.createEmptyDetachedField();
+				const src = { parent, parentField, parentIndex: start };
+				const dst = { parent: this.root, parentField: fieldKey, parentIndex: 0 };
```

Comment:
prefer whole word names, like source and destination.

Reply:
I have a habbit of using src and dst too, so there are a lot of them laying around. Feel free to leave this for now, and/or fix all of them in this file. It doesn't really matter too much, but we are supposed to favor full words (for both style and code accessability reasons)